### PR TITLE
made OkHttpClient a shared instance in MediaRestClient and MediaXMLRPCClient

### PR DESCRIPTION
**Pull request by mzorz**
Supersedes #399 and #400 (bad commit history)

This PR makes the OkHttpClient being used for media uploads a singleton, as recommended in OkHttpClient specs here https://square.github.io/okhttp/3.x/okhttp/okhttp3/OkHttpClient.html

> OkHttpClients should be shared
> 
> OkHttp performs best when you create a single OkHttpClient instance and reuse it for all of your HTTP calls. This is because each client holds its own connection pool and thread pools. Reusing connections and threads reduces latency and saves memory. Conversely, creating a client for each request wastes resources on idle pools.
> 

cc @aforcier

### CODE REVIEW

File: example/src/debug/java/org/wordpress/android/fluxc/example/DebugOkHttpClientModule.java [hunk 1]
```diff
@@ -41,4 +44,26 @@
         builder.addNetworkInterceptor(new StethoInterceptor());
         return builder;
     }
+
+    @Singleton
+    @Provides
+    @Named("custom-ssl")
+    public OkHttpClient provideOkHttpClientSharedInstanceCustomSSL(@Named("custom-ssl")OkHttpClient.Builder builder) {
```
Comment (maxme) [inline example/src/debug/java/org/wordpress/android/fluxc/example/DebugOkHttpClientModule.java]: I think we need to improve our checkstyle configuration here `@Named("custom-ssl") OkHttpClient.Builder builder`

File: fluxc/src/main/java/org/wordpress/android/fluxc/module/ReleaseOkHttpClientModule.java [hunk 2]
```diff
@@ -38,4 +41,26 @@
         }
         return builder;
     }
+
+    @Singleton
+    @Provides
+    @Named("custom-ssl")
+    public OkHttpClient provideOkHttpClientSharedInstanceCustomSSL(@Named("custom-ssl")OkHttpClient.Builder builder) {
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/module/ReleaseOkHttpClientModule.java]: This naming isn't very descriptive in this context. Maybe `provideMediaOkHttpClientInstance`? (Same for the method below and for the debug module.)

File: example/src/debug/java/org/wordpress/android/fluxc/example/DebugOkHttpClientModule.java (hunk 1 above)
Comment (aforcier) [inline example/src/debug/java/org/wordpress/android/fluxc/example/DebugOkHttpClientModule.java]: 👍 was thinking that too.

Comment (aforcier) [inline example/src/debug/java/org/wordpress/android/fluxc/example/DebugOkHttpClientModule.java]: I added this checkstyle rule in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/414. I say we leave this as is here and that PR will force us to add the space.

**Comment by aforcier** [comment]
This should be good to go except for a merge conflict and [this renaming nit](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/410#pullrequestreview-32614387).

File: fluxc/src/main/java/org/wordpress/android/fluxc/module/ReleaseOkHttpClientModule.java (hunk 2 above)
Comment (mzorz) [inline fluxc/src/main/java/org/wordpress/android/fluxc/module/ReleaseOkHttpClientModule.java]: addressed in f33fe65

**Comment by mzorz** [comment]
Ready for another go after Travis does its thing @aforcier 🙇

**Comment by aforcier** [comment]
@mzorz The changes in f33fe65 should also be done in `ReleaseOkHttpClientModule` 😉.

**Comment by mzorz** [comment]
> The changes in f33fe65 should also be done in ReleaseOkHttpClientModule 😉

🙇  addressed here 98b9ddb

File: fluxc/src/main/java/org/wordpress/android/fluxc/module/ReleaseOkHttpClientModule.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
 
     @Provides
     @Named("custom-ssl")
-    public OkHttpClient.Builder provideOkHttpClientBuilderCustomSSL(MemorizingTrustManager memorizingTrustManager) {
+    public OkHttpClient.Builder provideMediaOkHttpClientInstanceCustomSSL(MemorizingTrustManager memorizingTrustManager) {
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/module/ReleaseOkHttpClientModule.java]: It looks like these were renamed wrong - this one shouldn't be touched, it's not related to media. The renames need to shift down one method 😉.

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**